src/stores/app.store.ts

```diff
@@ -6,14 +6,29 @@ export const useAppStore = defineStore(
   () => {
     const lang = ref('fr')
     const layersOpen = ref()
+    const styleEditorOpen = ref(false)
+
     function setLang(language: string) {
       lang.value = language
     }
+
     function setLayersOpen(open: boolean) {
       layersOpen.value = open
     }
 
-    return { lang, layersOpen, setLang, setLayersOpen }
+    function toggleStyleEditorPanel(force?: boolean) {
+      styleEditorOpen.value =
+        force !== undefined ? force : !styleEditorOpen.value
```

Comment:
This could perhaps be simplified with a `setStyleEditorOpen(open: boolean)` method that can be called with `setStyleEditorOpen(!styleEditorOpen)` for toggling. This is what I did for `setLayersOpen()` to keep the code in the store as simple as possible. What do you think?

Reply:
Thanks
I will even split into 2 functions: toggle and close, then it is explicit.
since the function is called from html, it is better not having complicated function signatures.